nixos/modules/services/databases/neo4j.nix

```
@@ -493,6 +498,7 @@ in {
           revokedDir = mkOption {
             type = types.path;
             default = "${config.baseDirectory}/revoked";
+            defaultText = literalExpression ''"''${baseDirectory}/revoked"'';
```

With `options` added to the submodule arguments:

```suggestion
            defaultText = literalExpression ''"''${${options.baseDirectory}}/revoked"'';
```

This will then show as

```
"${services.neo4j.ssl.policies.<name>.baseDirectory}/revoked"
```

---

> Do you think you could use a patch like this to simplify many of these changes?

we had actually thought about doing *exactly that*, but then decided against it out of fear that it'll just make it impossible to get this merged. if you think it's wortwhile doing though, all for it!

---

at this point we should note that #147441 should also be redone for consistency, so we'd rather do it all in another PR than change this one.

---

I think it's very worthwhile indeed! For this PR it's kind of optional but still nice, however for https://github.com/NixOS/nixpkgs/pull/97023 it will be essential to have.

---

agreed! we can totally spin another PR that updates all the changes we've made so far. could run parallel to #149532, rebasing for those changes is easy enough.

---

On one hand I agree that this can be done in a separate PR. But on the other hand, doing this in the same one will be much easier regarding merge conflicts and making sure nothing was forgotten.

---

then we'll update this one to use option string coercions, and open another one later to update what #147441 did.